client/my-sites/plugins-wpcom/jetpack-plugins-panel.jsx

```diff
@@ -0,0 +1,220 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import {
+	identity,
+	find
+} from 'lodash';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import PluginIcon from 'my-sites/plugins/plugin-icon/plugin-icon';
+import Button from 'components/button';
+import Gridicon from 'components/gridicon';
+import JetpackPluginItem from './jetpack-plugin-item';
+import SectionHeader from 'components/section-header';
+import SectionNav from 'components/section-nav';
+import NavTabs from 'components/section-nav/tabs';
+import NavItem from 'components/section-nav/item';
+import Search from 'components/search';
+import jetpackPlugins from './jetpack-plugins';
+
+class JetpackPluginsPanel extends Component {
+
+	static propTypes = {
+		siteSlug: PropTypes.string,
+		hasBusiness: PropTypes.bool,
+		hasPremium: PropTypes.bool,
+	};
+
+	static defaultProps = {
+		translate: identity,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			selectedGroup: 'all',
+			searchTerm: '',
+		};
+	}
+
+	onNavClick( group ) {
+		this.setState( {
+			selectedGroup: group,
+		} );
+	}
+
+	onNavSearch = searchTerm => {
+		this.setState( {
+			searchTerm,
+		} );
+	}
+
+	getSearchPlaceholder() {
+		const { translate } = this.props;
+		switch ( this.state.selectedGroup ) {
+			case 'all':
+				return translate( 'Search All…', { textOnly: true } );
+			case 'engagement':
+				return translate( 'Search Engagement…', { textOnly: true } );
+			case 'security':
+				return translate( 'Search Security…', { textOnly: true } );
+			case 'appearance':
+				return translate( 'Search Appearance…', { textOnly: true } );
+			case 'writing':
+				return translate( 'Search Writing…', { textOnly: true } );
+		}
+	}
+
+	getSelectedText() {
+		const found = find( this.getNavItems(), item => this.state.selectedGroup === item.key );
+		return found ? found.title : '';
+	}
+
+	getNavItems() {
+		const { translate } = this.props;
+
+		return [
+			{
+				key: 'all',
+				title: translate( 'All', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'all' ),
+			},
+			{
+				key: 'engagement',
+				title: translate( 'Engagement', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'engagement' ),
+			},
+			{
+				key: 'security',
+				title: translate( 'Security', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'security' ),
+			},
+			{
+				key: 'appearance',
+				title: translate( 'Appearance', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'appearance' ),
+			},
+			{
+				key: 'writing',
+				title: translate( 'Writing', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'writing' ),
+			},
```

Comment:
we can eliminate the need to create new closures for `onClick` if we use custom attributes

```js
const navItems = {
	all: translate( 'All', { … } ),
	engagement: …
};

onNavClick = ( { target: { dataset: { groupName: selectedGroup } } } ) => this.setState( { selectedGroup } );

map( navItems, ( title, name ) => (
	<span data-group-name={ name } onClick={ this.onNavClick } />
) );
```